### PR TITLE
[@mantine/dates] DatePicker: Add dropdownPosition prop

### PR DESCRIPTION
**Pull request by victorlrz**
[V5] Feat https://github.com/mantinedev/mantine/issues/1636

- Add dropdownPosition prop to DatePicker and documentation

Link to [Master MR](https://github.com/mantinedev/mantine/pull/1638)

### CODE REVIEW

File: src/mantine-dates/src/components/DatePickerBase/DatePickerBase.tsx [hunk 1]
```diff
@@ -276,14 +280,15 @@ export const DatePickerBase = forwardRef<HTMLInputElement, DatePickerBaseProps>(
...
-          position="bottom-start"
           __staticSelector={__staticSelector}
           withinPortal={withinPortal}
           offset={10}
           opened={dropdownOpened}
           transitionDuration={transitionDuration}
           transition={transition}
           positionDependencies={positionDependencies}
+          middlewares={{ flip: dropdownPosition === 'flip', shift: false }}
+          position={dropdownPosition === 'flip' ? 'bottom' : dropdownPosition}
```
Comment (rtivital) [inline src/mantine-dates/src/components/DatePickerBase/DatePickerBase.tsx:291]: With these changes dropdown will be displayed at the bottom/top center while previously it was displayed at bottom/top start (`bottom-start` position)

**Review by rtivital — CHANGES_REQUESTED** [review]

File: src/mantine-dates/src/components/DatePickerBase/DatePickerBase.tsx (hunk 1 above)
Comment (Antyspi) [reply src/mantine-dates/src/components/DatePickerBase/DatePickerBase.tsx]: Should be fix, just added bottom-start and top-start, should we add others FloatingPosition ?

Comment (victorlrz) [reply src/mantine-dates/src/components/DatePickerBase/DatePickerBase.tsx]: Should be fixed, not sure if we add others FloatingPositions

**Comment by rtivital** [comment]
Thanks!

**Pull request merged**